lib/active_model/serializer/reflection.rb

```diff
@@ -78,6 +78,17 @@ def build_association(subject, parent_serializer_options)
 
       private
 
+      def polymorphic_key(assoc_value)
+        reflection_key = assoc_value.class.model_name
+        assoc_value.respond_to?(:each) ? reflection_key.plural : reflection_key.singular
```

Comment:
Cool, tell me whether this worked! (usage is `has_one :addressable, polymorphic: true`)